### ISSUE TITLE
Incorrect point cloud statistics

### ISSUE THREAD

**Issue by Brent-Edwards** [qgis/QGIS#51264]
### What is the bug or the crash?

Incorrect statistics reported in the "Statistics" section of the Layer Properties dialog. Suggestion to make the statistics accuracy message/warning more prominent.

### Steps to reproduce the issue

The statistics reported in the "Statistics" section of the Layer Properties dialog are incorrect (see image below). There is indeed a warning explaining that the stats are an estimate and thus warning the user of potential inaccuracies.

![image](https://user-images.githubusercontent.com/13538726/208478051-828a497d-509b-48ec-98bd-7a60fc575c27.png)

PDAL info call with actual counts:

![image](https://user-images.githubusercontent.com/13538726/208478790-7f85e707-015f-476b-a680-4330980d680d.png)

### Versions

QGIS version
3.28.1-Firenze
QGIS code revision
fde3b8fbb8c
Qt version
5.15.3
Python version
3.9.5
GDAL/OGR version
3.6.0
PROJ version
9.1.0
EPSG Registry database version
v10.074 (2022-08-01)
GEOS version
3.11.1-CAPI-1.17.1
SQLite version
3.39.4
PDAL version
2.4.3
PostgreSQL client version
unknown
SpatiaLite version
5.0.1
QWT version
6.1.6
QScintilla2 version
2.13.1
OS version
Windows 10 Version 2009

Active Python plugins
basemesh
2.0.11
gmsh
version 0.2.2
LAStools
1.4
openlayers_plugin
2.0.0
valuetool
3.0.15
wbt_for_qgis
1.0.5
db_manager
0.1.20
grassprovider
2.12.99
otbprovider
2.12.99
processing
2.12.99

### Supported QGIS version

- [X] I'm running a supported QGIS version according to the roadmap.

### New profile

- [ ] I tried with a new QGIS profile

**Comment by Brent-Edwards** [comment]
Could PDAL be used to get actual counts and avoid confusion?

**Comment by wonder-sk** [comment]
Yes we could do actual counts (and it does not really matter if we use PDAL for that or not). The main problem is that it is potentially quite an expensive operation to traverse all the points (e.g. if you have a 50 GB dataset streamed from a remote server), so it would need to be upon click of a button (and show progress bar, cancellation etc)
